Add docs for Postgres standalone setup

mmomjian opened the pull request:
This is a common source of issues with installations. Hoping to start consolidating the commands required in this article to have a place to point people to.

@mertalev appreciate your eyes

mmomjian committed:
Create postgres-standalone.md

mmomjian committed:
Update postgres-standalone.md

mmomjian committed:
Update postgres-standalone.md

mmomjian committed:
Update postgres-standalone.md

mmomjian committed:
Update postgres-standalone.md

mmomjian committed:
Update postgres-standalone.md

mmomjian committed:
Update postgres-standalone.md

mertalev commented on docs/docs/administration/postgres-standalone.md:
```diff
@@ -0,0 +1,45 @@
+# Preparing up a standalone Postgres server
```
It'd be helpful to have an intro here explaining what this page is about.

mertalev replied:
```suggestion
# Preparing a pre-existing Postgres server
```
I think "pre-existing" is more descriptive here. It's unclear what "standalone" means without more context.

mertalev commented on docs/docs/administration/postgres-standalone.md:
```diff
@@ -0,0 +1,45 @@
+
+While not the officially recommended method, it is possible to run Immich using an already-existing Postgres server by specifying `DB_URL` in the `.env` file.
+
+:::note
+By default, Immich expects superuser permission on the Postgres database.
+:::
+
+:::tip
+This method is recommended for users that have some baseline level of familiarity with Postgres and the Linux command line. If you do not have these, we recommend using the default setup with a separate Postgres container.
+:::
```
These can probably go in the intro.

mertalev replied:
It would also be clearer to replace "separate" with "dedicated" here.

mertalev commented on docs/docs/administration/postgres-standalone.md:
```diff
@@ -0,0 +1,45 @@
+
+Immich can run with a standalone Postgres server, without superuser permissions, by following the below instructions at the `psql` prompt to prepare the database.
+
+:::tip
+Before running these commands, you will need to install the version of vecto.rs as currently used by Immich to your system. You will also need to add `shared_preload_libraries = 'vectors.so'` to your `postgresql.conf`
+:::
```
This can be a "prerequisites" section instead since it's required.

mertalev replied:
Also, it should be noted that if they already have an entry for `shared_preload_libraries`, they can add `vectors.so` to it, comma-separated without spaces.

mertalev replied:
It would also be helpful to link to their [installation instructions](https://docs.pgvecto.rs/getting-started/installation.html) here.

mertalev commented on docs/docs/administration/postgres-standalone.md:
````diff
@@ -0,0 +1,45 @@
+# DB_URL='postgresql://immichdbusername:immichdbpassword@postgreshost:postgresport/immichdatabasename?sslmode=require&sslmode=no-verify'
+```
+
+## Standalone Postgres without superuser permissions
````
```suggestion
## Without superuser permissions
```

mertalev commented on docs/docs/administration/postgres-standalone.md:
```diff
@@ -0,0 +1,45 @@
+
+## Standalone Postgres without superuser permissions
+
+Immich can run with a standalone Postgres server, without superuser permissions, by following the below instructions at the `psql` prompt to prepare the database.
```
```suggestion
Immich can run without superuser permissions by following the below instructions at the `psql` prompt to prepare the database.
```

mertalev commented on docs/docs/administration/postgres-standalone.md:
```diff
@@ -0,0 +1,45 @@
+
+## Specifying the connection URL
+
+While not the officially recommended method, it is possible to run Immich using an already-existing Postgres server by specifying `DB_URL` in the `.env` file.
```
```suggestion
While not officially recommended, it is possible to run Immich using a pre-existing Postgres server by specifying `DB_URL` in the `.env` file.
```

mertalev commented on docs/docs/administration/postgres-standalone.md:
````diff
@@ -0,0 +1,45 @@
+Before running these commands, you will need to install the version of vecto.rs as currently used by Immich to your system. You will also need to add `shared_preload_libraries = 'vectors.so'` to your `postgresql.conf`
+:::
+
+```sql title="Setup Postgres for Immich"
````
```suggestion
```sql title="Set up Postgres for Immich"
```

mertalev reviewed (commented):
Thanks for working on this! We were actually just talking about making a page for this.

mmomjian committed:
Update docs/docs/administration/postgres-standalone.md

Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>

mmomjian committed:
Update docs/docs/administration/postgres-standalone.md

Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>

mmomjian committed:
Update docs/docs/administration/postgres-standalone.md

Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>

mmomjian committed:
Update docs/docs/administration/postgres-standalone.md

Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>

mmomjian committed:
Update docs/docs/administration/postgres-standalone.md

Co-authored-by: Mert <101130780+mertalev@users.noreply.github.com>

mmomjian committed:
Update postgres-standalone.md

mmomjian commented:
> Thanks for working on this! We were actually just talking about making a page for this.

Sure thing! I think I made all those updates, lmk what you think.

mmomjian committed:
Update postgres-standalone.md

Planning to write a guide in the future about setting up streaming database backups

mmomjian committed:
Update postgres-standalone.md

mmomjian commented:
@mertalev just tested this. I don't think the pg_vector_index_stat is created initially. I also tried adding the extension outside of a transaction block, and had the same behavior. We might want to try the ALTER DEFAULT method? 

```
trash1=# BEGIN;
BEGIN
trash1=*# CREATE EXTENSION vectors;
CREATE EXTENSION earthdistance CASCADE;
CREATE EXTENSION
NOTICE:  installing required extension "cube"
CREATE EXTENSION
trash1=*# \d
Did not find any relations.
trash1=*# GRANT SELECT ON TABLE pg_vector_index_stat to postgres;
ERROR:  relation "pg_vector_index_stat" does not exist
trash1=!# COMMIT;
ROLLBACK
```

mmomjian committed:
Update postgres-standalone.md

mmomjian committed:
Update postgres-standalone.md

mmomjian committed:
Update postgres-standalone.md

mertalev reviewed (approved):
Looks good!

mertalev commented:
That's interesting. We can go with `ALTER DEFAULT` if it doesn't exist immediately.
